staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -256,43 +249,18 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			metrics.WatchEvents.WithContext(req.Context()).WithLabelValues(kind.Group, kind.Version, kind.Kind).Inc()
 			isWatchListLatencyRecordingRequired := shouldRecordWatchListLatency(event)
 
-			if err := s.EmbeddedEncoder.Encode(event.Object, buf); err != nil {
-				// unexpected error
-				utilruntime.HandleError(fmt.Errorf("unable to encode watch object %T: %v", event.Object, err))
-				return
-			}
-
-			// ContentType is not required here because we are defaulting to the serializer
-			// type
-			unknown.Raw = buf.Bytes()
-			event.Object = &unknown
```

Comment:
this is mutating the event received ... doesn't that imply the event here is not shared between watchers (if it was shared, this mutation is not safe, right?)

if it is not shared, what is the benefit of caching the encoding in the event instance?

Reply:
Yes - events are not shared, but what's we're doing here [as you noticed in the other comment] is that we're actually caching the serialization in the object itself [which is shared across events].